src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/postponementrequest/PostponementRequestAboutToStartHandler.java

```diff
@@ -0,0 +1,51 @@
+package uk.gov.hmcts.reform.sscs.ccd.presubmit.postponementrequest;
+
+import static java.util.Objects.requireNonNull;
+import static org.apache.commons.collections4.CollectionUtils.emptyIfNull;
+
+import java.time.LocalDateTime;
+import java.util.Optional;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.sscs.ccd.callback.Callback;
+import uk.gov.hmcts.reform.sscs.ccd.callback.CallbackType;
+import uk.gov.hmcts.reform.sscs.ccd.callback.PreSubmitCallbackResponse;
+import uk.gov.hmcts.reform.sscs.ccd.domain.*;
+import uk.gov.hmcts.reform.sscs.ccd.presubmit.PreSubmitCallbackHandler;
+
+@Service
+public class PostponementRequestAboutToStartHandler implements PreSubmitCallbackHandler<SscsCaseData> {
+
+    @Override
+    public boolean canHandle(CallbackType callbackType, Callback<SscsCaseData> callback) {
+        requireNonNull(callback, "callback must not be null");
+        requireNonNull(callbackType, "callbacktype must not be null");
+
+        return callbackType.equals(CallbackType.ABOUT_TO_START)
+                && callback.getEvent() == EventType.POSTPONEMENT_REQUEST
+                && callback.getCaseDetails() != null
+                && State.HEARING.equals(callback.getCaseDetails().getState());
```

Comment:
Do you need to check the state here? Isn't that just controlled in the CCD definition?

Reply:
Yes, you're right. It is controlled by the ccd definition. We don't need the extra check here